libraries/AP_Proximity/AP_Proximity_Cygbot_D1.cpp

```diff
@@ -0,0 +1,185 @@
+#include "AP_Proximity_Cygbot_D1.h"
+
+#if HAL_PROXIMITY_ENABLED
```

Comment:
add AP_PROXIMITY_CYGBOT_ENABLED
and default it in AP_Proximity_Cygbot_D1.h based on HAL_PROXIMITY_ENABLED